flow/scenarios/multi_loop.py

```diff
@@ -0,0 +1,309 @@
+            see flow/core/params.py
+        num_vehicles : int
+            number of vehicles to be placed on the network
+        kwargs : dict
+            extra components, usually defined during reset to overwrite initial
+            config parameters
+
+        Returns
+        -------
+        startpositions : list of tuple (float, float)
+            list of start positions [(edge0, pos0), (edge1, pos1), ...]
+        startlanes : list of int
+            list of start lanes
+        """
+        (x0, min_gap, bunching, lanes_distr, available_length,
+         available_edges, initial_config) = \
+            self._get_start_pos_util(initial_config, num_vehicles, **kwargs)
+
+        increment = available_length / num_vehicles
+        vehs_per_ring = num_vehicles / self.num_rings
+
+        x = x0
+        car_count = 0
+        startpositions, startlanes = [], []
+
+        # generate uniform starting positions
+        while car_count < num_vehicles:
+            # collect the position and lane number of each new vehicle
+            pos = self.get_edge(x)
+
+            # place vehicles side-by-side in all available lanes on this edge
+            for lane in range(min([self.num_lanes(pos[0]), lanes_distr])):
+                car_count += 1
+                startpositions.append(pos)
+                edge, pos = startpositions[-1]
+                startpositions[-1] = edge, pos % self.length
+                startlanes.append(lane)
+
+                if car_count == num_vehicles:
+                    break
+
+            # 1e-13 prevents an extra car from being added in wrong place
+            x = (x + increment + VEHICLE_LENGTH + min_gap) + 1e-13
+            if (car_count % vehs_per_ring) == 0:
+                # if we have put in the right number of cars,
+                # move onto the next ring
+                ring_num = int(car_count / vehs_per_ring)
+                x = self.length * ring_num + 1e-13
+
+        # add a perturbation to each vehicle, while not letting the vehicle
+        # leave its current edge
+        # if initial_config.perturbation > 0:
+        #     for i in range(num_vehicles):
+        #         perturb = np.random.normal(0, initial_config.perturbation)
+        #         edge, pos = startpositions[i]
+        #         pos = max(0, min(self.edge_length(edge), pos + perturb))
+        #         startpositions[i] = (edge, pos)
+
+        return startpositions, startlanes
+
+    def specify_nodes(self, net_params):
+        """See parent class."""
+        length = net_params.additional_params["length"]
+        ring_num = net_params.additional_params["num_rings"]
+
+        r = length / (2 * pi)
+        ring_spacing = 4 * r
+        num_rows = num_cols = int(ceil(sqrt(ring_num)))
+
+        nodes = []
+        i = 0
+        for j in range(num_rows):
+            for k in range(num_cols):
+                nodes += [{
+                    "id": "bottom_{}".format(i),
```

Comment:
many double quotes. want to fix now or in a later commit?

Reply:
Later commit; I'll do one 100% only double quotes PR so that you don't even have to read it to approve.

Reply:
sounds good!